pkg/kube/kube_handler_test.go

```diff
@@ -1098,37 +1097,54 @@ func TestValidateAppRepository(t *testing.T) {
 		expectedHeaders            http.Header
 		expectedError              error
 		expectedReqResolutionError error
+		err                        error
+		response                   *http.Response
+		expectedResult             *ValidationResponse
 	}{
 		{
 			name:             "it parses the repo URL",
 			requestNamespace: kubeappsNamespace,
 			requestData:      `{"appRepository": {"name": "test-repo", "repoURL": "http://example.com/test-repo"}}`,
 			expectedURLs:     []string{"http://example.com/test-repo/index.yaml"},
+			err:              nil,
+			response:         &http.Response{StatusCode: 401, Body: ioutil.NopCloser(bytes.NewReader([]byte("Boom")))},
+			expectedResult:   &ValidationResponse{Code: 401, Message: "Boom"},
```

Comment:
So I *think* what you've done here is tried to combined the previous two sets of tests so that it now tests the handler in one go. Perhaps this could work, but it looks like as a result that you've had to avoid testing a positive case (below it's  always return a 401 as the fake) and delete a critical part of the tests for the expected requests and headers (seems to be what the diff says below, let me know if I've misread it).

What I was suggesting in the previous review was that you could update the existing test case (this current set) to test that the correct validator is returned with the correct data (ie. testing getValidator only, just like this set of tests was previously testing `getRequests` only). Something like:

```
func TestGetValidator(t *testing.T) {
    testCases := []struct {
        name string
        requestData string
        expectedValidator *HttpValidator
        expectedError
...
```
and all it needs to do is verify that the correct validator is returned with the correct data, including the request with headers for the non-OCI, or the correct AppRepository for the OCI case.

Then the second set of tests which test the actual validation (previously `doValidation()`, now `IsValid()`) could be done separately (ie. one set of tests for the non-oci case, and one set for the oci case, testing the two different implementations independently).

Have a go, but let me know if my description above is getting too hard to follow and I'll instead propose a diff showing you what I mean. I can also show a small refactoring that would allow testing the handler itself too.

Reply:
I have updated the test code to restore the old test functionality.